lib/ae_mdw/oracles.ex

```diff
@@ -0,0 +1,118 @@
+defmodule AeMdw.Oracles do
+  @moduledoc """
+  Context module for dealing with Oracles.
+  """
+
+  require AeMdw.Db.Model
+
+  alias AeMdw.Db.Format
+  alias AeMdw.Db.Model
+  alias AeMdw.Mnesia
+  alias AeMdw.Node
+
+  @type cursor :: binary()
+  # TODO: This needs to be an actual type like AeMdw.Db.Oracle.t()
+  @type oracle :: term()
+
+  @limit 10
+
+  @table_active AeMdw.Db.Model.ActiveOracle
+  @table_active_expiration Model.ActiveOracleExpiration
+  @table_inactive AeMdw.Db.Model.InactiveOracle
+  @table_inactive_expiration Model.InactiveOracleExpiration
+
+  @spec fetch_active_oracles(Mnesia.direction(), cursor() | nil) :: {[oracle()], cursor()}
+  def fetch_active_oracles(direction, cursor) do
+    {exp_keys, next_cursor} =
+      Mnesia.fetch_keys(
+        @table_active_expiration,
+        direction,
+        deserialize_cursor(cursor),
+        @limit
+      )
+
+    {:ok, {last_gen, -1}} = Mnesia.last_key(AeMdw.Db.Model.Block)
+
+    oracles =
+      exp_keys
+      |> Enum.map(fn {_expiration, oracle_pk} -> fetch_active_oracle(oracle_pk) end)
+      |> serialize_list(last_gen, true)
+
+    {oracles, serialize_cursor(next_cursor)}
+  end
+
+  @spec fetch_inactive_oracles(Mnesia.direction(), cursor() | nil) :: {[oracle()], cursor()}
+  def fetch_inactive_oracles(direction, cursor) do
+    {exp_keys, next_cursor} =
+      Mnesia.fetch_keys(
+        @table_inactive_expiration,
+        direction,
+        deserialize_cursor(cursor),
+        @limit
+      )
+
+    {:ok, {last_gen, -1}} = Mnesia.last_key(AeMdw.Db.Model.Block)
+
+    oracles =
+      exp_keys
+      |> Enum.map(fn {_expiration, oracle_pk} -> fetch_inactive_oracle(oracle_pk) end)
+      |> serialize_list(last_gen, false)
```

Comment:
nitpick: to use serialize terminology for cursors and render for maps to be encoded as json

Reply:
Good call, `render` for user-friendly structures and `serialize` for encoding/decoding into other structures